homework/index.js

```diff
@@ -30,17 +32,134 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function renderError(error) {
+    const container = document.getElementById('container');
+    container.innerHTML = '';
+    createAndAppend('div', container, {
+      text: error.message,
+      class: 'error',
+    });
+  }
+
+  function showRepositoryInfo(repository) {
+    const leftHand = document.getElementById('left-hand');
+    leftHand.innerHTML = '';
+    const tableElement = createAndAppend('table', leftHand);
+    const rowForName = createAndAppend('tr', tableElement);
```

Comment:
You have repeated code here: create a `tr` with two `td` children. You make your code DRYer, by creating a function, say `addRow`, that you could call like this:

```js
addRow(tbody, labelText, value);
```

For the first row (containing the `a` tag) you could do a 'fix up' after calling `addRow`:

```js
const firstRow = addRow(tbody, 'Repository', '');
createAndAppend('a', firstRow.lastChild, { href: ..., text: ... });
```

Reply:
Hello Jim Cramer!
Thanks for your review and notes. I want summarize your notes, if it's true please confirm it. 
1. I have to change class: error; to class: alert-error; 
2. When application starts automatically first repository will be selected and details will be shown on the page.
3. On the eventlistener, i should use onselect instead of onclick.
4. I have to create and use addRow() function for creating table rows.
Is that correct?